x-pack/plugin/mapper-constant-keyword/src/test/java/org/elasticsearch/xpack/constantkeyword/mapper/SearchIdleTests.java

```diff
@@ -0,0 +1,275 @@
+        // GIVEN
+        final String idleIndex = "test1";
+        final String activeIndex = "test2";
+        // NOTE: we need many shards because shard pre-filtering and the "can match" phase
+        // are executed only if we have enough shards.
+        int idleIndexShardsCount = 3;
+        int activeIndexShardsCount = 3;
+        assertAcked(createIndex(idleIndex, idleIndexShardsCount));
+        assertAcked(createIndex(activeIndex, activeIndexShardsCount));
+        assertAcked(
+            createIndexMapping(idleIndex, "constant_keyword", "type=constant_keyword,value=constant_value1", "keyword", "type=keyword")
+        );
+        assertAcked(
+            createIndexMapping(activeIndex, "constant_keyword", "type=constant_keyword,value=constant_value2", "keyword", "type=keyword")
+        );
+
+        assertEquals(RestStatus.CREATED, client().prepareIndex(idleIndex).setSource("keyword", randomAlphaOfLength(10)).get().status());
+        assertEquals(RestStatus.CREATED, client().prepareIndex(activeIndex).setSource("keyword", randomAlphaOfLength(10)).get().status());
+        assertEquals(RestStatus.OK, client().admin().indices().prepareRefresh(idleIndex, activeIndex).get().getStatus());
+
+        waitUntil(
+            () -> Arrays.stream(client().admin().indices().prepareStats(idleIndex, activeIndex).get().getShards())
+                .allMatch(ShardStats::isSearchIdle),
+            2,
+            TimeUnit.SECONDS
+        );
+
+        final IndicesStatsResponse idleIndexStatsBefore = client().admin().indices().prepareStats("test1").get();
+        Arrays.stream(idleIndexStatsBefore.getShards()).forEach(shardStats -> assertTrue(shardStats.isSearchIdle()));
+        Arrays.stream(idleIndexStatsBefore.getShards()).forEach(shardStats -> assertTrue(shardStats.getSearchIdleTime() >= 100));
+
+        final IndicesStatsResponse activeIndexStatsBefore = client().admin().indices().prepareStats("test2").get();
+        Arrays.stream(activeIndexStatsBefore.getShards()).forEach(shardStats -> assertTrue(shardStats.isSearchIdle()));
+        Arrays.stream(activeIndexStatsBefore.getShards()).forEach(shardStats -> assertTrue(shardStats.getSearchIdleTime() >= 100));
+
+        // WHEN
+        final SearchResponse searchResponse = search("test*", "constant_keyword", "constant_value2", 5);
+        assertEquals(RestStatus.OK, searchResponse.status());
+        assertEquals(idleIndexShardsCount, searchResponse.getSkippedShards());
+        assertEquals(0, searchResponse.getFailedShards());
+        Arrays.stream(searchResponse.getHits().getHits()).forEach(searchHit -> assertEquals("test2", searchHit.getIndex()));
+
+        // THEN
+        final IndicesStatsResponse idleIndexStatsAfter = client().admin().indices().prepareStats(idleIndex).get();
+        Arrays.stream(idleIndexStatsAfter.getShards()).forEach(shardStats -> assertTrue(shardStats.isSearchIdle()));
```

Comment:
Maybe assert that `idleIndexStatsAfter.getShards()` has elements or isn't empty?

Reply:
Maybe turn this then into a static helper method? So that we do this for other similar assertions too.